components/Document/DocumentHeaderV2.tsx

```diff
@@ -86,18 +87,42 @@ const DocumentHeader = ({ document: doc, metadata }: Props) => {
   }, []);
 
   const options: Array<MenuOption> = [
-    // ...(isPaper(doc)
-    //   ? [
-    //       {
-    //         label: "Edit metadata",
-    //         icon: <FontAwesomeIcon icon={faPen} />,
-    //         value: "edit-metadata",
-    //         onClick: () => {
-    //           alert("implement me");
-    //         },
-    //       },
-    //     ]
-    //   : []),
+    ...(isPaper(doc)
```

Comment:
Hmm.. doing checks like this looks already off. I just looked at what the function is doing, and it looks like it's just type-checking. 
If we are going to consolidate different documents to be viewed with a single component like this (which I like) shouldn't we have a consolidate payload from the backend. I believe BE already has to do something like this before it sends data to FE. 
In that case, might as well just put off this cumbersome parsing all off to BE instead of parsing twice in BE & FE.

Reply:
I'll share some thought.

We have `Paper` and `Post` types which are a little different from each other. Each of these extends `GenericDocument` so essentially a Post is a document and also Paper.

A paper for instance is defined as such:

```
export type Paper = GenericDocument & {
  journal?: string;
  isOpenAccess: boolean;
  laymanTitle: string;
  externalUrl?: string;
  abstract?: string;
  abstractHtml?: TrustedHTML;
  license?: string;
};
```

In the above case, since we don't know what type document is, we need to use `isPaper` which makes an assertion and avoids TS errors.

```
export const isPaper = (document: GenericDocument): document is Paper => {
  return (document as Paper).type === "paper";
};
```

The main takeaway above is that the return value is `document is Paper`